### ISSUE TITLE
[NBug] External program returned non-zero exit code.

### ISSUE THREAD

**Issue by jmongard** [gitextensions/gitextensions#12094]
### Environment

- Git Extensions 5.1.1.17970
- Build 939c0edba68f0c15b8e5180ca06a92c5ce534501
- Git 2.47.0.windows.2
- Microsoft Windows NT 10.0.19045.0
- .NET 8.0.11
- DPI 96dpi (no scaling)
- Portable: False

### Issue description

Exit code: 1
Command: C:\Program Files\Git\bin\git.exe
Arguments: verify-tag --raw test-tag
Working directory: C:\Dev\src\my-server\

```
GitExtensions.Extensibility.ExternalOperationException: External program returned non-zero exit code.
 ---> System.Exception: External program returned non-zero exit code.
   --- End of inner exception stack trace ---
   at async Task<ExecutionResult> GitCommands.ExecutableExtensions.ExecuteAsync(IExecutable executable, ArgumentString arguments, Action<StreamWriter> writeInput, Encoding outputEncoding, CommandCache cache, string extraCacheKey, bool stripAnsiEscapeCodes, bool throwOnErrorExit, CancellationToken cancellationToken) in C:/projects/gitextensions/src/app/GitCommands/Git/ExecutableExtensions.cs:line 338
   at void Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at T Microsoft.VisualStudio.Threading.JoinableTask<T>.CompleteOnCurrentThread()
   at T Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run<T>(Func<Task<T>> asyncMethod, JoinableTaskCreationOptions creationOptions) x 2
   at ExecutionResult GitCommands.ExecutableExtensions.Execute(IExecutable executable, ArgumentString arguments, Action<StreamWriter> writeInput, Encoding outputEncoding, CommandCache cache, bool stripAnsiEscapeCodes, bool throwOnErrorExit, CancellationToken cancellationToken) in C:/projects/gitextensions/src/app/GitCommands/Git/ExecutableExtensions.cs:line 266
   at string GitCommands.Gpg.GitGpgController.GetTagVerificationMessage(IGitRef tagRef, bool raw) in C:/projects/gitextensions/src/app/GitCommands/Git/GitGpgController.cs:line 247
   at Task<TagStatus> GitCommands.Gpg.GitGpgController.GetRevisionTagSignatureStatusAsync(GitRevision revision)+() => { } in C:/projects/gitextensions/src/app/GitCommands/Git/GitGpgController.cs:line 170
   at async Task<TagStatus> GitCommands.Gpg.GitGpgController.GetRevisionTagSignatureStatusAsync(GitRevision revision) in C:/projects/gitextensions/src/app/GitCommands/Git/GitGpgController.cs:line 161
   at async Task<GpgInfo> GitUI.CommandsDialogs.FormBrowseController.LoadGpgInfoAsync(GitRevision revision) in C:/projects/gitextensions/src/app/GitUI/CommandsDialogs/FormBrowseController.cs:line 32
   at async Task GitUI.CommandsDialogs.FormBrowse.FillGpgInfoAsync(GitRevision revision) in C:/projects/gitextensions/src/app/GitUI/CommandsDialogs/FormBrowse.cs:line 1290
   at async void GitUI.TaskManager.InvokeAndForget(Control control, Func<Task> asyncAction, CancellationToken cancellationToken)+(?) => { } [1] in C:/projects/gitextensions/src/app/GitExtUtils/GitUI/TaskManager.cs:line 119
   at async Task GitUI.TaskManager.HandleExceptionsAsync(Func<Task> asyncAction, Func<Exception, Task> handleExceptionAsync) in C:/projects/gitextensions/src/app/GitExtUtils/GitUI/TaskManager.cs:line 32
```

If you add an unsigned annotated tag to a signed commit the GPG-tab will show an exception.

### Steps to reproduce

1. create a signed commit.
2. show the GPG-tab
3. create an unsigned annotated tag on the signed commit.
4. exception

### Diagnostics

The command that git extensions try to execute is:

```bash
git verify-tag --raw "the-tag"
```

git responds with: `error: no signature found` (exit code: 1)

**Comment by RussKie** [comment]
The app is showing the non-success result (exit code 1) from the underlying git command. The dialog can be safely dismissed.

As for the question - should the dialog not being shown - I don't know, afterall it's a failing command (i.e., the exit code != 0).

**Comment by mstv** [comment]
Duplicate of #12048, resolved in #12049.
